deepspeed/inference/engine.py

```diff
@@ -162,10 +162,10 @@ def __init__(self,
             torch.cuda.set_rng_state(_rng_state.cpu())
 
         if self.mp_world_size > 1:
+            assert not self.enable_cuda_graph, "Cuda graph is not supported for model parallelism"
+        elif self.enable_cuda_graph:
             self.model_orig_fwd = self.module.forward
             self.module.forward = self.forward
-        else:
-            self.module.register_forward_pre_hook(self._pre_forward_hook)
```

Comment:
The forward hook was initially added to make sure that all tensors are on GPU! But I think it is just safe to remove it as it is the responsibility of user to put them on device!